ophyd/controls/areadetector/filestore_mixins.py

```diff
@@ -0,0 +1,167 @@
+"""Mixin classes that customize the filestore integration of AreaDetector
+FilePlugins.
+
+To be used like so:
+
+    from ophyd.controls.areadetector.detectors import PerkinElmerDetector
+    from ophyd.controls.areadetector.plugins import HDF5Plugin
+    from ophyd.controls.areadetector.trigger_mixins import SingleTrigger
+    from ophyd.controls.areadetector.filestore_mixins import (
+        FileStoreIterativeWrite)
+
+    class MyPlugin(HDF5Plugin, FileStoreIterativeWrite):
+        pass
+
+    dest = '/tmp'  # in production, use a directory on your system -- not /tmp
+
+    class MyDetector(PerkinElmerDetector, SingleTrigger):  # for example
+        file_plugin = MyPlugin(suffix='HDF1:', write_file_path=dest)
+
+    det = MyDetector(...)
+"""
+
+from __future__ import print_function
+import logging
+import uuid
+import filestore.api as fs
+
+from datetime import datetime
+from collections import defaultdict, OrderedDict
+from itertools import count
+import os
+
+from ..ophydobj import DeviceStatus
+from ..device import GenerateDatumInterface, BlueskyInterface
+
+logger = logging.getLogger(__name__)
+
+
+def new_uid():
+    "uuid4 as a string"
+    return str(uuid.uuid4())
+
+
+def new_short_uid():
+    "uuid4, skipping the last stanza because of AD length restrictions."
+    return '-'.join(new_uid().split('-')[:-1])
+
+
+class FileStoreBase(BlueskyInterface, GenerateDatumInterface):
+    "Base class for FileStore mixin classes"
+    def __init__(self, *args, write_file_path=None, read_file_path=None,
```

Comment:
It's a minor thing, really. It's just that if you expect the logic shown in the initializer to be consistent, it may surprise you:

```python
class Detector(Device):
    p = Cpt(FilePlugin, write_file_path='/tmp')

det1 = Detector()
# ok, good

det2 = Detector()
det2.p.write_file_path = '/something_new'
# but det2.p.read_file_path = '/tmp'
```